aicsimageio/formats.py

```diff
@@ -194,4 +195,7 @@
     "lfr": "aicsimageio.readers.default_reader.DefaultReader",
     "lfp": "aicsimageio.readers.default_reader.DefaultReader",
     "spe": "aicsimageio.readers.default_reader.DefaultReader",
+
+    "nd2": "aicsimageio.readers.bioformats_reader.BioformatsReader",
```

Comment:
what does this look like when the full implementation is ready? It's making me imagine things like BioformatsReader.supported_extensions : List[str]

Reply:
I think we are still waiting for a decision / conclusion on: https://github.com/AllenCellModeling/aicsimageio/issues/307

Reply:
see possible endings (though note that's not always sufficient) which can also be introspected with Java: http://j.mp/bio-formats-list

Reply:
We should figure this out before this is merged imo. Does anyone on the review list have hard opinions on the resolution of #307... @tlambert03 do you have hard opinions on that one? I haven't tested this reader out at all but in your testing of it what were your gripes.

Reply:
And regardless of what our resolution strategy is, all of the formats for bioformats should be unpacked to this list before merge.

Reply:
> do you have hard opinions on that one?

not really.  probably leaning slightly to the one-reader-per-extension as @toloudis suggested.  Since ultimately that list is consumed by the AICSImage object (right?) having the additional overhead of a fallback reader seems excessive – at least at first – since a user could still directly indicate the backend if they wanted to override, (correct?)

Reply:
Ahhhh true. I forgot that the `reader=` param exists on the `AICSImage` object so having one reader per format is just a totally fine default.

Reply:
> I haven't tested this reader out at all but in your testing of it what were your gripes.

main gripe is probably the time to load (at first).  Then it speeds up a bit.  but for things that it attempts, it feels pretty robust.  that is: as I started playing with other files, they tended to "just work" without any special considerations.  which I guess is how it "should" be given that most of this is tested on the bioformats side?

Reply:
When merging / accepting this PR, I don't think we would consider it "experimental" at all.  There may or may not be use cases that are non optimal or don't work but by including it we are basically saying we think the 80% case is working well.  The intent would be to support it moving forward.

Reply:
And let me ALSO say a thanks to @tlambert03 here.  This is a very high impact update.  

The very early versions of this library (well prior to it arriving on github, and also pre-daskification) tried to use python-bioformats from CellProfiler and we ran into lots of issues internally with multiple java vms running at the same time, crashes etc.   We (AICS) felt that a more "pure" python solution (resulting in using tifffile and czifile inside the Readers in those earliest versions) was going to be more robust and that was the case for us.  

But now is the perfect time to add this support.  We do still prefer more "native" python format readers but having this type of universal fallback is just perfect -- so also big thanks to the OME for BioFormats even existing.  As I said in a prior discussion, we now have something where before there was nothing.